srtcore/group.cpp

```diff
@@ -4468,18 +4468,18 @@ void CUDTGroup::updateLatestRcv(CUDTSocket* s)
 
     UniqueLock lg(m_GroupLock);
     // Sanity check for a case when getting a deleted socket
-    if (!s->m_IncludedGroup)
+    if (!s->m_GroupOf)
         return;
 
     // Under a group lock, we block execution of removal of the socket
-    // from the group, so if m_IncludedGroup is not NULL, we are granted
-    // that m_IncludedIter is valid.
-    gli_t current = s->m_IncludedIter;
+    // from the group, so if m_GroupOf is not NULL, we are granted
+    // that m_GroupMemberData is valid.
+    SocketData* current = s->m_GroupMemberData;
 
     for (gli_t gi = m_Group.begin(); gi != m_Group.end(); ++gi)
     {
         // Skip the socket that has reported packet reception
-        if (gi == current)
+        if (&*gi == current)
```

Comment:
What is the purpose of dereferencing and again taking address here?

Reply:
Because `gi` is an iterator, not a pointer.

Reply:
Right, `m_Group` holds `std::list<SocketData>`, not `std::list<SocketData*>`.